File: cpp/leet_code/1700.cpp
// Optimizing the code for speed and unrolling loops for efficiency.
#pragma GCC optimize("O3,unroll-loops")
#include <vector>

class Solution {
public:
  int countStudents(std::vector<int>& students, std::vector<int>& sandwiches) {
    int count[2] = {0, 0};
    for (int i = 0; i < students.size(); i++) {
      count[students[i]]++;
    }
    for (int i = 0; i < sandwiches.size(); i++) {
      if (count[sandwiches[i]] == 0) {
        break;
      }
      count[sandwiches[i]]--;
    }
    return count[0] + count[1];
  }
};

File: rust/leet_code/src/1700.rs
impl Solution {
    pub fn count_students(students: Vec<i32>, sandwiches: Vec<i32>) -> i32 {
        let mut students = students;
        let mut sandwiches = sandwiches;
        let mut i = 0;
        while i < students.len() {
            if students[i] == sandwiches[0] {
                students.remove(i);
                sandwiches.remove(0);
                i = 0;
            } else {
                i += 1;
            }
        }
        students.len() as i32
    }
}
